hyperspy/_signals/hologram_image.py

```
@@ -0,0 +1,274 @@
+                              'HologramImage automatically.')
+                reference.set_signal_type('hologram')
+            elif reference is not None:
+                reference = HologramImage(reference)
+
+        # Parsing sideband position:
+        if sb_position is None:
+            warnings.warn('Sideband position is not specified. The sideband will be found automatically which may '
+                          'cause wrong results.')
+            if reference is None:
+                sb_position = self.find_sideband_position(sb=sb)
+            else:
+                sb_position = reference.find_sideband_position(sb=sb)
+
+        if self.axes_manager.navigation_size:
+            if sb_position.ndim == 1:
+                sb_position = np.stack([sb_position]*self.axes_manager.navigation_size)
+
+            elif sb_position.ndim == 2 and sb_position.shape[0] != self.axes_manager.navigation_size:
+                warnings.warn('The sideband position size does not match the size of the hologram image! '
+                              'The first values of sideband position will be used to reconstruct all slices'
+                              'of hologram image.')
+                sb_position = np.stack([sb_position[0]]*self.axes_manager.navigation_size)
+
+        if sb_size is None:  # Default value is 1/2 distance between sideband and central band
+            sb_size = self.find_sideband_size(sb_position)
+
+        folded = self.unfold_navigation_space()
+
+        # Standard edge smoothness of sideband aperture 5% of sb_size
+        if sb_smooth is None:
+            sb_smooth = sb_size * 0.05
+
+        # Convert sideband size from 1/nm or mrad to pixels
+        if sb_unit == 'nm':
+            sb_size /= np.mean(self.f_sampling)
+            sb_smooth /= np.mean(self.f_sampling)
+        elif sb_unit == 'mrad':
+            try:
+                ht = self.metadata.Acquisition_instrument.TEM.beam_energy
+            except AttributeError:
```

This should instead raise an error explaining that the user have to define the beam energy first using `set_microscope_parameters`.